src/main/webapp/scripts/serverclient.js

```diff
@@ -0,0 +1,79 @@
+import { SessionCache } from './sessioncache';
+import { Session } from './session';
+
+/**
+ * ServerClient is responsible for keeping up-to-date with the current 
+ * session and handles many of the client-to-server interactions, 
+ * like passing the controller.
+ */
+class ServerClient {
+  /**
+   * Initalizes a ServerClient object.
+   * @param {URLSearchParams} urlParams Represents the URLSearchParams 
+   *    of the session the client is in, holds information such as the
+   *    session ID and the screen name of the current user.
+   */
+  constructor(urlParams) {
+    /**
+     * function sessionRequest_() is the fetch api request
+     * responsible for contacting the server to retrieve the Session
+     * object.
+     * @return {function(): Promise<any>} 
+     * @private
+     */
+    async function sessionRequest_() {
+      const /** string */ name = encodeURI(urlParams.get('name'));
+      const /** string */ sessionID = 
+          encodeURI(urlParams.get('session-id'));
+      const /** Response */ response = await fetch(
+          `/get-session-info?name=${name}&session-id=${sessionID}`);
+      return await response.json();
+    }
+
+    /**
+     * The SessionCache for the current session, contacts the server for 
+     * the Session object.
+     * @private {SessionCache}
+     */
+    this.sessionCache_ = new SessionCache(sessionRequest_);
+
+    /**
+     * @private {URLSearchParams}
+     */
+    this.urlParams_ = urlParams;
+  }
+
+  /** 
+   * This method starts to cache.
+   */
+  start() {
+    throw new Error('Unimplemented');
+  }
+
+  /** 
+   * This method stops caching.
+   */
+  stop() {
+    throw new Error('Unimplemented');
+  }
```

Comment:
These should be private. The user of the client should not bear the burden of knowing these internal details, if it doesnt have to

Reply:
so should start just be called in the constructor?

Reply:
Mhm. Call start in the constructor. Not sure where you would call stop(). Honestly, there probably isnt even a reason to ever call stop() lol i guess you can listen for the webpage closing but that seems extra. Lets leave stop() out for now